src/fight.cpp

```diff
@@ -395,17 +395,30 @@ void set_fighting(struct char_data * ch, struct char_data * vict, ...)
     }
   }
   if (!already_there) {
-    ch->next_fighting = combat_list;
-    combat_list = ch;
+    // Setting init to 0 here means new combat actually starts with the next global re-roll.
+    // This prevents arbitray length out-of-initiative states as new combats are initiated.
+    // Also need to avoid being first in the list to avoid setting off an early global re-roll.
+    GET_INIT_ROLL(ch) = 0;
+    ch->next_fighting = combat_list->next_fighting;
```

Comment:
This assumes that the combat list is populated, and will crash if nobody is in combat at the time it runs (`combat_list` = `NULL`).

Reply:
Oh, right.